crates/gen-wasmtime/src/lib.rs

```diff
@@ -1881,27 +1881,29 @@ impl Bindgen for FunctionBindgen<'_> {
                     let tmp = self.tmp();
                     self.push_str(&format!("let ptr{} = {};\n", tmp, operands[0]));
                     self.push_str(&format!("let len{} = {};\n", tmp, operands[1]));
-                    let result = format!(
+                    self.push_str(&format!(
                         "
-                                copy_slice(
-                                    &mut caller,
-                                    memory,
-                                    func_{},
-                                    ptr{tmp}, len{tmp}, {}
-                                )?
-                            ",
-                        free,
+                            let data{tmp} = copy_slice(
+                                &mut caller,
+                                memory,
+                                ptr{tmp}, len{tmp}, {}
+                            )?;
+                        ",
                         align,
-                        tmp = tmp
+                        tmp = tmp,
+                    ));
+                    self.call_intrinsic(
+                        free,
+                        format!("(ptr{tmp}, len{tmp} as i32, {})", align, tmp = tmp),
```

Comment:
Ah I remember now why I bundled the free in copy_slice, the `len` here is incorrect since it needs to be the byte size instead of the item-size. I think that should be reasonably easy to fix though with a multiplication here (we know it shouldn't overflow else the `copy_slice` would have failed)